### PR TITLE
Test incomplete stream reads in Alluxio caching

### PR DESCRIPTION
**Pull request by jkylling**
## Description

Adds tests coverage for the bug fixed in b2df2f3f8ef312527ebb1a0b3ce2f09e2aa91ea1

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                throws IOException
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public int read(byte[] buff, int off, int len)
+                throws IOException
+        {
+            return delegate.read(buff, off, len / 2 + len % 2);
```
Comment (wendigo) [inline lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java:110]: can we go random here? :)

Comment (jkylling) [reply lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java]: Yes

**Review by wendigo — APPROVED** [review]

File: lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    static class IncompleteTrinoInputStream
+            extends TrinoInputStream
+    {
+        private final TrinoInputStream delegate;
+        private final Random random;
+
+        public IncompleteTrinoInputStream(TrinoInputStream delegate)
+        {
+            this.delegate = delegate;
+            this.random = new Random(42);
```
Comment (wendigo) [inline lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java:99]: 🪄

File: lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        @Override
+        public int read()
+                throws IOException
+        {
```
[the remark this answers is not in the document]
Comment (jkylling) [reply lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/IncompleteStreamTrinoFileSystem.java]: Just laziness. Changed it to delegate

**Pull request merged**